### DISCUSSION TITLE
How to use loader without permitNullValues (i want to keep this as false)

### DISCUSSION THREAD

**Discussion by Mukund2900** [cache2k/cache2k#197] [Q&A]
I am using loader for read through operations, but I dont want to save null values inside my cache. 
When I try to implement this use-case it throws an internal server error/ basically saying that null values are not permitted in the cache. I want it to just return null and not save the value at all in this case. Cant find any such case explained in the documentation. Can someone help?  
```
cache = new Cache2kBuilder<String, Map<String, String>>() {
        }
                 .name(cacheName)
                .eternal(true)
                .entryCapacity(2000)
                .loader(key -> expensiveOperation())
                .build();
```

this expensive operation returns null at times. Is there a way to not save that in the cache and still return value null when I do a cache.get(key);

**Comment by globalworming** [comment]
Please check [how-to-enable-null-values](https://cache2k.org/docs/latest/user-guide.html#how-to-enable-null-values) and [loader-and-null-values](https://cache2k.org/docs/latest/user-guide.html#loader-and-null-values)
It seems to me that you would need to permit null values and then set an expiry policy that immediately expires null values. Given you don't want to cache null values

**Reply by Mukund2900** [reply]
I know how to enable null values @globalworming but whats the whole point of saving a null value when we dont want it in our cache at all. Should there not be a direct method which does not save null values and returns null when the API/expensive method operation fails. 
If that makes any sense. 
Expiring a null value immediately doesnt make sense. (more like a crack)

**Reply by globalworming** [reply]
i do admit it feels like a workaround and a `permitNullValues = true` and a hypothetical `cacheNullValues = false` would feel better. But it is what the documentation is saying.
